docs/specification/draft/client/user_interaction.mdx

```diff
@@ -0,0 +1,355 @@
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 123,
+  "method": "interaction/create",
+  "params": {
+    "id": "c8ec9a71-c475-4a88-a774-e9cd154bc7f8",
+    "type": "ua",
+    "interaction": {
+      "url": "https://example.com/mcp?state=1234567890&...",
+      "message": {
+        "type": "text",
+        "text": "Please sign in to your account."
+      }
+    }
+  }
+}
+```
+
+#### Client Response
+
+When the client facilitates the opening of the URL, it **MUST** send a response to the server:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 123,
+  "result": {}
+}
+```
+
+Alternatively, if the end-user cancels or declines the interaction, the client **MUST**
+send an error response to the server:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 123,
+  "error": {
+    "code": -32099,
+    "message": "User did not consent to the interaction."
+  }
+}
+```
+
+## Cancelling an Interaction
+
+The MCP server **MAY** cancel an interaction at any time by sending a
+`notifications/cancelled` message to the client, as described in [Cancellation](/specification/draft/basic/utilities/cancellation).
+The MCP client **SHOULD NOT** send a response.
+
+
+## Security Considerations
+
+### Handling Sensitive Information
+
+MCP servers **MUST** use the `ua` interaction type for sensitive information, such as
+auth credentials or authorization grants. The MCP client **MUST NOT** send sensitive
+information to the MCP server in any other way.
+
+### User Consent
+
+The MCP client **SHOULD** clearly communicate the purpose of the interaction to the end-user,
+and provide a way for the end-user to consent to the interaction before continuing.
+
+### Identifying the User
+
+The MCP server **MUST NOT** rely on input from the client or end-user to identify the
+end-user, as this untrusted input can be forged and lead to impersonation attacks.
+
+Instead, the MCP server **SHOULD** identify the end-user in trusted (server) code. For
+example:
+
+- Implementations using an HTTP transport and [MCP authorization](/specification/draft/basic/authorization)
+  **MAY** use the subject claim (`sub`) from the OAuth 2.1 Access Token as the user
+  identifier, if present.
+- Implementations using an stdio transport **MAY** acquire the user identifier from the
+  environment.
+
+### Server-side Request Forgery
+
+With the `ua` interaction type, the MCP client is expected to make a request via a user agent
+based on the value of the `url` parameter in the interaction object. Since MCP enables clients
+to interoperate with MCP servers it has no prior knowledge of, this opens a risk for Server-Side
+Request Forgery (SSRF) attacks by malicious MCP servers. MCP clients **SHOULD** take appropriate
+precautions against SSRF attacks, such as blocking requests to internal IP address ranges and
+requiring TLS.
+
+Additionally, the MCP client **SHOULD** validate the `url` parameter in the interaction object and
```

Comment:
This needs some detail as to what "validation" actually means here.

Reply:
👍 I'll flesh this out